repository_cli/cli/records.py

```diff
@@ -382,66 +382,94 @@ def replace_identifier(identifier: str, pid: str):
+    obj = None
+    try:
+        obj = files[key]
+    except Exception as e:
+        click.echo("File does not yet exist.")
 
-    obj = files[key]
     if obj is not None and not replace_existing:
-        click.echo(click.style(u'File with key "{key}" already exists.'
-                   u' Use `--replace-existing/-f` to overwrite it.'.format(
-                        key=key, recid=recid), fg='red'))
+        click.echo(
+            click.style(
+                'File with key "{key}" already exists.'
+                " Use `--replace-existing/-f` to overwrite it.".format(
+                    key=key, recid=recid
+                ),
+                fg="red",
+            )
+        )
         return
 
     fp.seek(SEEK_SET, SEEK_END)
     size = fp.tell()
     fp.seek(SEEK_SET)
 
-    click.echo(u'Will add the following file:\n')
-    click.echo(click.style(
-        u'  key: "{key}"\n'
-        u'  bucket: {bucket}\n'
-        u'  size: {size}\n'
-        u''.format(
-            key=key,
-            bucket=bucket.id,
-            size=size),
-        fg='green'))
-    click.echo(u'to record:\n')
-    click.echo(click.style(
-        u'  Title: "{title}"\n'
-        u'  RECID: {recid}\n'
-        u'  UUID: {uuid}\n'
-        u''.format(
-            recid=record['id'],
-            title=record.get('metadata', {}).get('title'),
-            uuid=record.id),
-        fg='green'))
+    click.echo("Will add the following file:\n")
+    click.echo(
+        click.style(
+            '  key: "{key}"\n'
+            "  bucket: {bucket}\n"
+            "  size: {size}\n"
+            "".format(key=key, bucket=bucket.id, size=size),
+            fg="green",
+        )
+    )
+    click.echo("to record:\n")
+    click.echo(
+        click.style(
+            '  Title: "{title}"\n'
+            "  RECID: {recid}\n"
+            "  UUID: {uuid}\n"
+            "".format(
+                recid=record["id"],
+                title=record.get("metadata", {}).get("title"),
+                uuid=record.id,
+            ),
+            fg="green",
+        )
+    )
     if replace_existing and obj is not None:
-        click.echo(u'and remove the file:\n')
-        click.echo(click.style(
-            u'  key: "{key}"\n'
-            u'  bucket: {bucket}\n'
-            u'  size: {size}\n'
-            u''.format(
-                key=obj.key,
-                bucket=bucket,
-                size=obj.file.size),
-            fg='green'))
-
-    if click.confirm(u'Continue?'):
+        click.echo("and remove the file:\n")
+        click.echo(
+            click.style(
+                '  key: "{key}"\n'
+                "  bucket: {bucket}\n"
+                "  size: {size}\n"
+                "".format(key=obj.key, bucket=bucket, size=obj.file.size),
+                fg="green",
+            )
+        )
+
+    if click.confirm("Continue?"):
+        files.enabled = True  # this allows to also add files to metadata only records
```

Comment:
Could you please create a check here and add a parameter to the command. It seams not a good idea of doing that without consent of the user who is running that command. It maybe intended, but it could also be a mistake to add a record to a metadata only record.

Reply:
added a check here: https://github.com/tu-graz-library/repository-cli/pull/30/commits/b71aaefc8ccfec6c7ac075b7dbe73c53b8172a78#diff-f5a1fe025043505f7d2f8fd63880616044dc1475637fde84827637497e61f7e3R428-R434 , so checks are together.